internal/webhook/external/github.go

```diff
@@ -0,0 +1,170 @@
+package external
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	gh "github.com/google/go-github/v71/github"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	xhttp "github.com/akuity/kargo/internal/http"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+// githubHandler handles push events for github.
+// After the request has been authenticated,
+// the kubeclient is queried for all warehouses that contain a subscription
+// to the repo in question. Those warehouses are then patched with a special
+// annotation that signals down stream logic to refresh the warehouse.
+func githubHandler(
+	c client.Client,
+	namespace string,
+	secretName string,
+) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		logger := logging.LoggerFromContext(ctx)
+		logger.Debug("retrieving secret",
+			"secret-name", secretName,
+		)
+		var secret corev1.Secret
+		err := c.Get(ctx,
+			client.ObjectKey{
+				Name:      secretName,
+				Namespace: namespace,
+			},
+			&secret,
+		)
+		if err != nil {
+			logger.Error(err, "failed to get github secret")
+			xhttp.WriteErrorJSON(w,
+				xhttp.Error(
+					fmt.Errorf("failed to get github secret %q: %w", secretName, err),
+					http.StatusNotFound,
```

Comment:
The problem here isn't that the path didn't route to anywhere. The problem is with server-side configuration (e.g. the server's problem; not the client's), so I would probably regard this as an internal server error instead of not found.